src/obvinf.cpp

```diff
@@ -0,0 +1,149 @@
+//
+// libsemigroups_pybind11
+// Copyright (C) 2024 James Mitchell
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+// libsemigroups headers
+#include <libsemigroups/cong.hpp>
+#include <libsemigroups/kambites.hpp>
+#include <libsemigroups/obvinf.hpp>
+#include <libsemigroups/todd-coxeter.hpp>
+
+// pybind11....
+#include <pybind11/pybind11.h>
+
+// libsemigroups_pybind11....
+#include "main.hpp"  // for init_obvinf
+
+namespace py = pybind11;
+
+namespace libsemigroups {
+
+  void init_obvinf(py::module& m) {
+    m.def(
+        "is_obviously_infinite",
+        [](Presentation<word_type> const& p) {
```

Comment:
Do we want to do this for `Presentation<std::string>` too?

Reply:
Good point forgot that one!